File: Python/juegosinherencia/Mago.py

class Mago:
    DANO_ESPADACHIN = 10
    DANO_LADRON = 20
    VIDA_POR_DEFECTO = 100

    def __init__(self, nombre):
        self._nombre = nombre
        self._vida = Mago.VIDA_POR_DEFECTO
    
    @classmethod    
    def Constructor(cls, nombre, vida):
        nuevoMago = cls(nombre, vida)
        return nuevoMago

    @property
    def nombre(self):
        return self._nombre

    @property
    def vida(self):
        return self._vida

    @vida.setter
    def vida(self, valor):
        if valor < 0:
            valor = 0
        self._vida = valor

    def recibir_dano(self, dano):
        self.vida -= dano

    def atacar_ladron(self, objetivo):
        print(f"{self.nombre} ataca a {objetivo.nombre}!")
        objetivo.recibir_dano(self.DANO_LADRON)
        print(f"{objetivo.nombre} tiene {objetivo.vida} puntos de vida restantes.\n")

    def atacar_espadachin(self, objetivo):
        print(f"{self.nombre} ataca a {objetivo.nombre}!")
        objetivo.recibir_dano(self.DANO_ESPADACHIN)
        print(f"{objetivo.nombre} tiene {objetivo.vida} puntos de vida restantes.\n")

File: Java/juegosinherencia/src/Mago.java
public class Mago {

    private String nombre;
    private int vida;

    private static final int DANO_ESPADACHIN = 10;
    private static final int DANO_LADRON = 20;
    private static final int VIDA_POR_DEFECTO = 100;

    public Mago(String nombre, int vida) {
        this.nombre = nombre;
        this.vida = vida;
    }

    public Mago(String nombre) {
        this(nombre, VIDA_POR_DEFECTO);
    }

    public String getNombre() {
        return nombre;
    }

    public int getVida() {
        return vida;
    }

    public void recibirDano(int dano) {
        vida -= dano;
        if (vida < 0)
            vida = 0;
    }

    public void atacar(Ladron objetivo) {
        System.out.println(getNombre() + " ataca a " + objetivo.getNombre() + "!");
        objetivo.recibirDano(DANO_LADRON);
        System.out.println(objetivo.getNombre() + " tiene " + objetivo.getVida() + " puntos de vida restantes.");
        System.out.println();
    }

    public void atacar(Espadachin objetivo) {
        System.out.println(getNombre() + " ataca a " + objetivo.getNombre() + "!");
        objetivo.recibirDano(DANO_ESPADACHIN);
        System.out.println(objetivo.getNombre() + " tiene " + objetivo.getVida() + " puntos de vida restantes.");
        System.out.println();
    }
}